usbmount.conf

```diff
@@ -46,8 +46,8 @@ MOUNTOPTIONS="noexec,nodev,noatime,nodiratime"
 # For example, "-fstype=vfat,gid=floppy,dmask=0007,fmask=0117" would add
 # the options "gid=floppy,dmask=0007,fmask=0117" when a vfat filesystem
 # is mounted.
-FS_MOUNTOPTIONS="-fstype=vfat,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=exfat,uid=1000,gid=1000,dmask=0007,fmask=0177 -fstype=ntfs,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8"
+FS_MOUNTOPTIONS="-fstype=vfat,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=ntfs,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=ext2,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=ext3,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=ext4,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=hfsplus,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=fuseblk,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=ntfs-3g,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=fuse,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=msdos,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=umsdos,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8 -fstype=fat,uid=1000,gid=1000,dmask=0007,fmask=0177,flush,utf8"
```

Comment:
I haven't gone through all of these yet, but not all of the file system types have the same mount options. For example, ntfs does not have dmask, fmask, or flush. I can go through these in more detail, but the file system specific mount options can be found [here](https://man7.org/linux/man-pages/man8/mount.8.html#:~:text=still%20work%20properly.-,FILESYSTEM%2DSPECIFIC%20MOUNT%20OPTIONS,-top). These need to be audited to optimize each file systems specific mount options.

Reply:
So it looks like the documentation I was looking at was old/incomplete. Here is as list of all of the file systems and their respective documentation: [](https://www.kernel.org/doc/Documentation/filesystems/)

Reply:
Looking through them now

Reply:
These are my notes through looking at the documentation. I also want to make sure as part of this ticket that we make sure that we are installing all of the necessary drivers for all of the file systems supported.

ntfs
 - no flush
 
ntfs-3g 
 - No flush
 - no utf8
 
I also found NTFS3 in the documentation, should we add this?
 
ext2
 - No uid or gid but does have grpid
 - No dmask or fmask
 - No flush
 - No utf8
 
ext3
 - Same as ext2
 
ext4
 - Same as ext3
 
hfsplus
  - No dmask or fmask
  - Has umask
  
Should we add hfs?

fuse and fuse-blk
 - No gid or uid
 - No dmask or fmask
 - No flush
 - No utf8
 - Has user_id and group_id
 
fat
 - No utf8
 - No flush
 
umsdos and msdos
 - Same as fat

Reply:
hfsplus also does not have flush